src/data/finset/lattice.lean

```diff
@@ -142,6 +142,14 @@ begin
     exact ⟨le_trans hay hyz, le_trans hsx_sup hxz⟩, },
 end
 
+-- If we acquire sublattices
+-- the hypotheses should be reformulated as `s : subsemilattice_inf_bot`
+lemma sup_mem
+  (s : set α) (w₁ : ⊥ ∈ s) (w₂ : ∀ x y ∈ s, x ⊔ y ∈ s)
+  {ι : Type*} (t : finset ι) (p : ι → α) (h : ∀ i, p i ∈ s) :
+  t.sup p ∈ s :=
+@sup_induction _ _ _ _ _ s w₁ w₂ (λ i _, h i)
```

Comment:
Is something wrong with the signature of `sup_induction` that makes these underscores needed?

Reply:
I think it's precisely the difference between `set X` and `X -> Prop`. If you make the `p` argument (`s` here) argument of `sup_induction` explicit then it is happy without the `@`. But as is it won't guess `s`, because it looks wrong.

Reply:
Are you sure it's not just that `sup_induction` is missing `@[elab_as_eliminator]`? It should be able to infer `p = (∈ s)` without having to know what `set X` unfolds to.

Reply:
I wonder why it doesn't need it for the second one.

Reply:
(also it might make sense to rename `p` to `f` in this lemma to match the other lemmas in this file)

Reply:
Doesn't seem to help.

Reply:
Yeah, I can't get it to work either. Let's avoid having to unfold `set` at least:
```suggestion
@sup_induction _ _ _ _ _ (∈ s) w₁ w₂ (λ i _, h i)
```